pennylane/_qubit_device.py

```diff
@@ -480,12 +480,12 @@ def marginal_prob(self, prob, wires=None):
 
         wires = Wires(wires)
         # get indices of wires on the device's register
-        registers = self.wire_map(wires)
+        wire_indices = self.indices_on_register(wires)
 
-        registers = np.hstack(registers)
+        wire_indices = np.hstack(wire_indices)
```

Comment:
Actually, I don't get what the hstack is supposed to do here (but didn't want to introduce unecessary changes). Since `wires_indices` is just a list of integers, this converts the list into an array.  @josh146 would you know if this was catering for the cases of nested wires before?

Reply:
It was, I believe! If you remove it and the tests pass, it _should_ be safe...